jobs/spFilteringJob.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/data-preservation-programs/spade-tenant/config"
+	"github.com/data-preservation-programs/spade-tenant/db"
+	"github.com/joho/godotenv"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"gorm.io/gorm"
+)
+
+type SP struct {
+	Version        int       `bson:"__v"`
+	Provider       string    `bson:"_id"`
+	AgentCity      string    `bson:"agentCity"`
+	AgentCountry   string    `bson:"agentCountry"`
+	AgentLatitude  float64   `bson:"agentLatitude"`
+	AgentLongitude float64   `bson:"agentLongitude"`
+	AgentRegion    string    `bson:"agentRegion"`
+	Date           time.Time `bson:"date"`
+	LatencyMs      float64   `bson:"latencyMs"`
+	Multiaddr      string    `bson:"multiaddr"`
+	TestId         string    `bson:"testId"`
+}
+
+func main() {
+
+	sps := setUpEnvAndFetchSpDataFromMongo()
+
+	config := config.InitConfig()
+	database, err := db.OpenDatabase(config.DB_URL, config.DEBUG, config.DRY_RUN)
+
+	if err != nil {
+		panic(err)
+	}
+
+	var clauses []db.TenantSPEligibilityClauses
+	database.Model(&db.TenantSPEligibilityClauses{}).Find(&clauses)
+	tenantClauseMap := createTenantClauseMap(clauses)
+
+	for tenantId, clauses := range tenantClauseMap {
+		for _, sp := range sps {
+			isValid := isValidClause(clauses, sp)
+			updateDatabase(isValid, database, sp, tenantId)
+		}
+	}
+}
+
+func setUpEnvAndFetchSpDataFromMongo() []SP {
+	godotenv.Load()
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	opts := options.Client().ApplyURI(os.Getenv("MONGODB_URI")).SetServerAPIOptions(serverAPI)
+
+	client, err := mongo.Connect(context.TODO(), opts)
+	if err != nil {
+		panic(err)
+	}
+
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	collection := client.Database(os.Getenv("MONGODB_DATABASE")).Collection(os.Getenv("MONGODB_SP_COLLECTION"))
+
+	cur, _ := collection.Find(context.TODO(), bson.D{{}})
+	var sps []SP = []SP{}
+	for cur.Next(context.TODO()) {
+		var sp SP
+		err := cur.Decode(&sp)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		sps = append(sps, sp)
+	}
+
+	return sps
+}
+
+func isValidClause(clauses []db.TenantSPEligibilityClauses, sp SP) bool {
```

Comment:
hmm, I'm a bit confused by the naming `isValid` and `isValidClause`, could this be more accurately named as `isEligible` and `verifyEligibility` ?

Reply:
good call out I think that is clearer as well